torchgeo/trainers/change.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for change detection."""
+
+import os
+import warnings
+from typing import Any, Optional, Union, cast
+
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.classification import (
+    MulticlassAccuracy,
+    MulticlassF1Score,
+    MulticlassJaccardIndex,
+)
+from torchmetrics.wrappers import ClasswiseWrapper
+from torchvision.models._api import WeightsEnum
+
+from ..models import FCSiamConc, FCSiamDiff, get_weight
+from . import utils
+from .base import BaseTask
+
+
+class FocalJaccardLoss(nn.Module):
+    """FocalJaccardLoss."""
+
+    def __init__(self) -> None:
+        """Initialize a FocalJaccardLoss instance."""
+        super().__init__()
+        self.focal_loss = smp.losses.FocalLoss(
+            mode="multiclass", normalized=True)
+        self.jaccard_loss = smp.losses.JaccardLoss(mode="multiclass")
+
+    def forward(self, preds: torch.Tensor, targets: torch.Tensor) -> torch.Tensor:
+        """Compute the loss."""
+        return cast(torch.Tensor, self.focal_loss(preds, targets) + self.jaccard_loss(preds, targets))
+
+
+class XEntJaccardLoss(nn.Module):
+    """XEntJaccardLoss."""
```

Comment:
Is there a paper ref for this?

Reply:
Not that I'm aware of, and I didn't find one from a quick search. This loss function was already in there with what I started from.